analytics/requirements.txt

```diff
@@ -1,11 +1,97 @@
-django==4.2.4
+amqp==5.2.0
+annotated-types==0.6.0
+argon2-cffi==23.1.0
+argon2-cffi-bindings==21.2.0
+asgiref==3.7.2
+asttokens==2.4.1
+billiard==4.2.0
+Brotli==1.1.0
+cachetools==5.3.2
+celery==5.3.6
+certifi==2023.11.17
+cffi==1.16.0
+chardet==5.2.0
+charset-normalizer==3.3.2
+click==8.1.7
+click-didyoumean==0.3.0
+click-plugins==1.1.1
+click-repl==0.3.0
+colorama==0.4.6
+decorator==5.1.1
+distlib==0.3.8
+dj-database-url==2.1.0
+dj-email-url==1.0.6
+Django==4.2.9
 django-click==2.3.0
+django-configurations==2.5
+django-cors-headers==4.3.1
+django-debug-toolbar==4.2.0
 django-extensions==3.2.3
-kubernetes==26.1.0
-python-gitlab==3.11.0
-psycopg2-binary==2.9.5
-opensearch-dsl==2.0.1
-sentry-sdk[django]
-gunicorn
-celery[redis]
-python-dateutil
+django-girder-utils==0.13.1
+django-minio-storage==0.5.7
+django-ninja==1.1.0
+django-s3-file-field==1.0.0
+djangorestframework==3.14.0
+executing==2.0.1
+factory-boy==3.3.0
+Faker==22.5.1
+filelock==3.13.1
+google-auth==2.26.2
+gunicorn==21.2.0
+idna==3.6
+iniconfig==2.0.0
+ipython==8.20.0
+jedi==0.19.1
+kombu==5.3.5
+kubernetes==29.0.0
+markdown-it-py==3.0.0
+matplotlib-inline==0.1.6
+mdurl==0.1.2
+minio==7.2.3
```

Comment:
Why do we have both a `requirements.txt` and a `setup.py` file?

Reply:
One specifies the concrete requirements and the other the abstract ones.

Reply:
And that's used for the prod django image? I'm a bit worrisome of having two independently managed lists of dependencies (i.e. if we change a dependency in `setup.py` and go to re-deploy, it will fail / not work unless we remember to update `requirements.txt`). Do you see this as an issue?

Reply:
I'll leave the settling of this up to you. If you don't think it's an issue or would prefer to address it separately, that's fine.